commit b552bb4445e124a65eb3fabe1690e301a3d95ff8
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:21 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/lamarck/patch-xinstall-have_setmode.diff

diff --git a/src/pextlib1.0/Pextlib.c b/src/pextlib1.0/Pextlib.c
index 6d826d716..148e54ccb 100644
--- a/src/pextlib1.0/Pextlib.c
+++ b/src/pextlib1.0/Pextlib.c
@@ -119,6 +119,8 @@ extern char **environ;
 
 #ifndef HAVE_SETMODE
 #include "setmode.h"
+#else
+#include <bsd/unistd.h>
 #endif
 
 __attribute__((format(printf, 3, 0)))
diff --git a/src/pextlib1.0/xinstall.c b/src/pextlib1.0/xinstall.c
index 2199a229c..a1863a484 100644
--- a/src/pextlib1.0/xinstall.c
+++ b/src/pextlib1.0/xinstall.c
@@ -116,6 +116,8 @@ extern int copyfile(const char *from, const char *to, void *state,
 
 #ifndef HAVE_SETMODE
 #include "setmode.h"
+#else
+#include <bsd/unistd.h>
 #endif
 
 /* Bootstrap aid - this doesn't exist in most older releases */
